dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -0,0 +1,635 @@
+ * @return the DataFrame containing the data from the SQL table.
+ */
+public fun DataFrame.Companion.readSqlTable(dbConfig: DatabaseConfiguration, tableName: String): AnyFrame {
+    DriverManager.getConnection(dbConfig.url, dbConfig.user, dbConfig.password).use { connection ->
+        return readSqlTable(connection, tableName, DEFAULT_LIMIT)
+    }
+}
+
+/**
+ * Reads data from an SQL table and converts it into a DataFrame.
+ *
+ * @param [dbConfig] the configuration for the database, including URL, user, and password.
+ * @param [tableName] the name of the table to read data from.
+ * @param [limit] the maximum number of rows to retrieve from the table.
+ * @return the DataFrame containing the data from the SQL table.
+ */
+public fun DataFrame.Companion.readSqlTable(dbConfig: DatabaseConfiguration, tableName: String, limit: Int): AnyFrame {
+    DriverManager.getConnection(dbConfig.url, dbConfig.user, dbConfig.password).use { connection ->
+        return readSqlTable(connection, tableName, limit)
+    }
+}
+
+/**
+ * Reads data from an SQL table and converts it into a DataFrame.
+ *
+ * @param [connection] the database connection to read tables from.
+ * @param [tableName] the name of the table to read data from.
+ * @return the DataFrame containing the data from the SQL table.
+ */
+public fun DataFrame.Companion.readSqlTable(connection: Connection, tableName: String): AnyFrame {
+    return readSqlTable(connection, tableName, DEFAULT_LIMIT)
+}
+
+/**
+ * Reads data from an SQL table and converts it into a DataFrame.
+ *
+ * @param [connection] the database connection to read tables from.
+ * @param [tableName] the name of the table to read data from.
+ * @param [limit] the maximum number of rows to retrieve from the table.
+ * @return the DataFrame containing the data from the SQL table.
+ */
+public fun DataFrame.Companion.readSqlTable(connection: Connection, tableName: String, limit: Int): AnyFrame {
+    var preparedQuery = "SELECT * FROM $tableName"
+    if (limit > 0) preparedQuery += " LIMIT $limit"
+
+    val url = connection.metaData.url
+    val dbType = extractDBTypeFromURL(url)
+
+    connection.createStatement().use { st ->
+        logger.debug { "Connection with url:${url} is established successfully." }
+        val tableColumns = getTableColumnsMetadata(connection, tableName)
+
+        st.executeQuery(
+            preparedQuery
+        ).use { rs ->
+            val data = fetchAndConvertDataFromResultSet(tableColumns, rs, dbType, limit)
+            return data.toDataFrame()
+        }
+    }
+}
+
+/**
+ * Converts the result of an SQL query to the DataFrame.
+ *
+ * @param [dbConfig] the database configuration to connect to the database, including URL, user, and password.
+ * @param [sqlQuery] the SQL query to execute.
+ * @return the DataFrame containing the result of the SQL query.
+ */
+public fun DataFrame.Companion.readSqlQuery(dbConfig: DatabaseConfiguration, sqlQuery: String): AnyFrame {
```

Comment:
Are there other options for sqlQuery type? Something like PreparedStatement, maybe? I'm afraid that with String it's tempting to use interpolation, and interpolation is prone to injections

Reply:
From other side, this give freedom to write what user wants, it's not a client library for Java Enterprise with UI elements and library user has its own responsibility to protect his database

I'm trying to avoid to expose low-level java.sql objects in the API, but probably it's not a possible.
Also I thought about parsing sqlQuery/tableName Strings with regular expression to be a valid table name (based on metadata) or valid SELECT - like query (without ; and DELETE/UPDATE sql keywords)

But it's a good idea to provide additional function with Statement or PreparedStatement